openedx_tagging/core/tagging/system_defined_taxonomies/object_tags.py

```diff
@@ -0,0 +1,210 @@
+    """
+
+    class Meta:
+        proxy = True
+
+    tag_class_model = None
+
+
+    def _check_taxonomy(self):
+        """
+        Validates if has an associated Django model that has an Id
+        """
+        if not super()._check_taxonomy():
+            return False
+    
+        if not self.tag_class_model:
+            return False
+
+        # Verify that is a Django model
+        if not issubclass(self.tag_class_model, models.Model):
+            return False
+
+        # Verify that the model has 'id' field
+        try:
+            self.tag_class_model._meta.get_field('id')
+        except FieldDoesNotExist:
+            return False
+
+        return True
+    
+    def _check_instance(self):
+        """
+        Validates if the instance exists
+        """
+        try:
+            intance_id = int(self.value)
+        except ValueError:
+            return False
+        return self.tag_class_model.objects.filter(id=intance_id).exists()
+
+    def _check_tag(self):
+        """
+        Validates if the instance exists
+        """
+        if not super()._check_tag():
+            return False
+        
+        # Validates if the instance exists
+        if not self._check_instance():
+            return False
+        
+        return True
+
+
+class UserObjectTag(ModelObjectTag):
+    """
+    Object tags used on taxonomies associated with user model
+    """
+
+    class Meta:
+        proxy = True
+
+    tag_class_model = get_user_model()
+
+
+class LanguageObjectTag(ClosedSystemObjectTag):
+    """
+    Object tag for Languages
+
+    The tags are filtered and validated taking into account the 
+    languages available in Django LANGUAGES settings var
+    """
+
+    system_defined_taxonomy_id = SystemDefinedIds.LanguageTaxonomy.value
+
+    class Meta:
+        proxy = True
+
+    @classmethod
+    def get_tags_query_set(cls, taxonomy: Taxonomy) -> models.QuerySet:
+        """
+        Returns a query set of available languages tags.
+        """
+        available_langs = cls._get_available_languages()
+        return taxonomy.tag_set.filter(external_id__in=available_langs)
+    
+    @classmethod
+    def _get_available_languages(cls) -> List[str]:
+        """
+        Get the available languages from Django LANGUAGE.
+        """
+        langs = set()
+        for django_lang in settings.LANGUAGES:
+            # Split to get the language part
+            langs.add(django_lang[0].split('-')[0])
+        return langs
+
+    def _check_tag(self):
+        """
+        Validates if the language tag is on the available languages
+        """
+        if not super()._check_tag():
+            return False
+
+        available_langs = self._get_available_languages()
+
+        # Must be linked to a tag and must be an available language
+        if not self.tag or not self.tag.external_id in available_langs:
+            return False
+        
+        return True
```

Comment:
Could even do this shorter as well, if you think it's legible enough:
```suggestion
        return (
            super()._check_tag() and 
            # Must be linked to a tag with an available language
            self.tag and self.tag.external_id in self._get_available_languages()
        )
```